lib/pbench/server/cache_manager.py

```diff
@@ -1423,16 +1431,22 @@ def __init__(self, options: PbenchServerConfig, logger: Logger):
         # resource_id.
         self.datasets: dict[str, Tarball] = {}
 
-    def full_discovery(self) -> "CacheManager":
+    def full_discovery(self, full: bool = True) -> "CacheManager":
         """Discover the ARCHIVE and CACHE trees
 
-        NOTE: both _discover_unpacked() and _discover_results() rely on the
-        results of _discover_archive(), which must run first.
+        Full discovery is only needed for reporting, and is not required
+        to find, create, or delete a specific dataset. (See find_dataset.)
 
-        Full discovery is not required in order to find, create, or delete a
-        specific dataset.
+        Args:
+            full: search the ARCHIVE tree rather than using Dataset table
+
+        Returns:
+            CacheManager instance for chaining
         """
-        self._discover_controllers()
+        if full:
+            self._discover_controllers()
+        else:
+            self._discover_datasets()
         return self
```

Comment:
Why not just replace
```
    cache_m.full_discovery(full=...)
```
with
```
    cache_m._discover_datasets()
```
or
```
    cache_m._discover_controllers()
```
as appropriate (with non-underscore-prefixed names, perhaps) rather than wrapping them?

Reply:
From the outside, this is two subtly different styles of full discovery, and I wanted to package them that way instead of inventing *two* new names for external entry points.